lib/blocks.php

```diff
@@ -172,6 +172,40 @@ function gutenberg_reregister_core_block_types() {
 
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
 
+/**
+ * Adds the defer loading strategy to all registered blocks.
+ *
+ * This function would not be part of core merge. Instead, the register_block_script_handle() function would be patched
+ * as follows.
+ *
+ * ```
+ * --- a/wp-includes/blocks.php
+ * +++ b/wp-includes/blocks.php
+ * @ @ -153,7 +153,8 @ @ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
+ *                 $script_handle,
+ *                 $script_uri,
+ *                 $script_dependencies,
+ * -           isset( $script_asset['version'] ) ? $script_asset['version'] : false
+ * +         isset( $script_asset['version'] ) ? $script_asset['version'] : false,
+ * +         array( 'strategy' => 'defer' )
```

Comment:
@westonruter, do we have a patch against WordPress core that sets the default strategy for `viewScripts` to `defer`?

Reply:
I'm catching up after an extended leave, so I might find the answer myself in the meantime as I'm reading the communication related to the `defer` strategy. I'm mostly willing to ensure we have better defaults in WP core, so people have performant configuration by design.

Reply:
Opened Core-59115 and https://github.com/WordPress/wordpress-develop/pull/5019

Reply:
Committed to core: [r56398](https://core.trac.wordpress.org/changeset/56398)

Reply:
Let's enable it for all blocks in both the Gutenberg plugin, and in WordPress core to get as much testing as necessary with 3rd party blocks that use `block.json`. I see that you discussed it in the context of view scripts in https://github.com/WordPress/gutenberg/pull/52536#discussion_r1273414888 and ruled out as a potential issue.

> Committed to core: [r56398](https://core.trac.wordpress.org/changeset/56398)

Noting that with this patch, `defer` is going to be used with `editorScript`, `script`, and `viewScript`. In the case of `editorScript`, the dependencies aren't using `defer` like `wp-block` or `wp-block-editor` so it will probably fallback to the legacy handling. I'll comment on Trac, too.

Reply:
> Let's enable it for all blocks in both the Gutenberg plugin, and in WordPress core to get as much testing as necessary with 3rd party blocks that use `block.json`. I see that you discussed it in the context of view scripts in [#52536 (comment)](https://github.com/WordPress/gutenberg/pull/52536#discussion_r1273414888) and ruled out as a potential issue.

Yes, it should be enabled for all blocks, whether core or 3rd party.

> Noting that with this patch, `defer` is going to be used with `editorScript`, `script`, and `viewScript`. In the case of `editorScript`, the dependencies aren't using `defer` like `wp-block` or `wp-block-editor` so it will probably fallback to the legacy handling. I'll comment on Trac, too.

Oh, good catch!